docs/source-app/workflows/byoc/index.rst

```diff
@@ -0,0 +1,105 @@
+#################################
+
+**Audience:** Users looking to run Lightning Apps on their own private cloud.
+
+----
+
+*******************
+A bit of background
+*******************
+
+BYOC - Bring Your Own Cloud, is an alternate deployment model to Lightning Cloud (fully managed SaaS).
+BYOC separates the control and data plane. The data plane, that includes
+Lightning clusters, services and Lightning Apps, reside inside the user’s VPC.
+The control plane resides on Lightning Cloud.
+
+Setup begins with configuring a cloud provider (today AWS, but more are coming soon) with your personal credentials for
+delegated access and an identity provider for secure access to the data plane.
+
+Next, as part of the environment creation process, you can configure networking,
+security, and select among cluster configuration options based on their own use cases.
+
+After submitting a cluster creation request, the Lightning Control Plane creates the required cloud infrastructure on the user account. This
+sets up a new Lightning Cluster along with a Lightning Kubernetes Operator.
+
+
+*******************************
+Create a Lightning BYOC cluster
+*******************************
+
+You must have your cloud configured before you try and create a BYOC cluster.
+
+And to make your life a little easier, we've made a `Terraform module to help with that <https://github.com/Lightning-AI/terraform-aws-lightning-byoc>`_.
+
+Create a Lightning BYOC cluster using the following command:
+
+.. code:: python
+
+   lightning create cluster <cluster-name> <cloud-provider-parameters>
+
+Here's an example:
+
+.. code:: python
+
+   lightning create cluster my-byoc-cluster --provider aws --role-arn arn:aws:iam::1234567890:role/lai-byoc --external-id dummy --region us-west-2 --instance-types t3.xlarge --enable-performance
+
+..note:: Cluster creation is gonna take an hour or more after you run this command.
+
+**Arguments**
+
+* cluster_name: The name of the cluster to be created
+
+.. note:: Cluster names can only contain lowercase letters, numbers, and periodic hyphens ( - ).
+
+**Parameters:**
+
+* provider: The cloud provider where your cluster is located.
+
+.. note:: AWS is supported today, but support for other cloud providers is coming soon.
+
+* role-arn: AWS IAM Role ARN used to provision resources
+
+* external-id: AWS IAM Role external ID
+
+.. note:: To read more on what the AWS external ID is and why it's useful go `here <https://docs.aws.amazon.com/IAM/latest/UserGuide/id_roles_create_for-user_externalid.html`_.
+
+* region: AWS region containing compute resources
+
+* instance-types: Instance types that you want to support, for computer jobs within the cluster. For now, this is the AWS instance types supported by the cluster.
+
+* enable-performance: Specifies if the cluster uses cost savings mode.
+
+.. note:: In cost saving mode the number of compute nodes is reduced to one, reducing the cost for clusters with low utilization.
```

Comment:
Note: too many of these notes will disturb the reader visually, because it will render a big blue box around it. I suggest to minimize this usage. In a bullet point list, you may also need to add indentation to make it render properly.

Reply:
So the trouble is, we want to add help at the point where it is needed. I could make this a table, but the notes will still be included, so I'm not sure how much ahead we will be.

Reply:
I've put the content into a table. It's a bit better.